### PR TITLE
fix: Session:single_pass isn't plan aware

### PR DESCRIPTION
**Pull request by shellz-n-stuff**
## Description
This PR fixes a minor oversight in the `single_pass` function where if a plan was provided to the session it would not be processed by the single_pass function.

This issue has been addressed by implementing a global "has_plan" check to maintain if there was originally a plan for the session. We also check the current length of the exchange message stack to ensure we perform this action in the event there was a plan added and it's not yet been processed.

Additionally, the initial_message variable has been changed to optional to allow CLI tools to take advantage of only leveraging session plans as prompts from Goose plugins.

### CODE REVIEW

File: src/goose/cli/session.py [hunk 1]
```diff
@@ -140,23 +141,30 @@ def process_first_message(self) -> Optional[Message]:
...
+            initial_message (Optional[str]): The initial user message to process.
         """
         profile = self.profile_name or "default"
         print(f"[dim]starting session | name: [cyan]{self.name}[/]  profile: [cyan]{profile}[/]")
         print(f"[dim]saving to {self.session_file_path}")
 
-        # Process initial message
-        message = Message.user(initial_message)
+        # Check to see if there is a planned operation to perform prior to the bespoke prompt
+        if self.has_plan and len(self.exchange.messages) > 0:
```
Comment (baxen) [inline src/goose/cli/session.py:159]: I might make initial_message and has_plan incompatible, checking for one or the other? 

related: if we have already loaded the plan, i don't think we need line 159, although it doesn't hurt either, more important is that we make sure we don't both add initial message and hte plan

**Review by baxen — APPROVED** [review]
LGTM!!

**Review by michaelneale — APPROVED** [review]
good catch

File: src/goose/cli/session.py (hunk 1 above)
Comment (shellz-n-stuff) [reply src/goose/cli/session.py:159]: I wanted to leave that open to the user/plugin developer to decide. Use cases in my mind were to support:
1. User wants to run a plan then ask it a question. 
2. User wants to just run plan
3. User wants to just ask a single question

This was the only way I could find to support all 3

**Pull request merged**